lib/srv/db/common/kerberos/client.go

```diff
@@ -0,0 +1,155 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package kerberos
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/jcmturner/gokrb5/v8/client"
+	"github.com/jcmturner/gokrb5/v8/config"
+	"github.com/jcmturner/gokrb5/v8/keytab"
+
+	"github.com/gravitational/teleport/lib/auth/windows"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/teleport/lib/srv/db/common/kerberos/kinit"
+)
+
+type clientProvider struct {
+	AuthClient windows.AuthInterface
+	DataDir    string
```

Comment:
We had some offline discussion on the cache dir. we can look into that afterwards. Here are just some notes but  they should not block this PR:

`kinit` does refresh the output every connection, but there could be a race that two connections trying to output to the same file then read it.

On the other handle, it seems `credentials.LoadCCache` finishes reading that file immediately and has no use of it afterwards. We could experiment if we could just use tmp dir instead of data-dir.

Another consideration is if we want to add real caching to avoid kinit on every connection.

Reply:
Another possibility is to reimplement the the protocol internally, without shelling out to `kinit`. Not only we would drop the external dependency and the need to juggle various files. As @gabrielcorado points out, we would also benefit from much better control over the whole process and much better diagnostic information.

Reply:
For reference here is the outstanding issue in `gokrb5`: 
- https://github.com/jcmturner/gokrb5/issues/158